src/index.js

```diff
@@ -5,14 +5,43 @@ import {
     addRenderedClassNames, getRenderedClassNames
 } from './inject';
 
+// TODO(emily): Make a 'production' mode which doesn't prepend the class name
+// here, to make the generated CSS smaller.
+const makeClassName = (key, vals) => `${key}_${hashObject(vals)}`;
+
+// Find all of the references to descendant styles in a given definition,
+// generates a class name for each of them based on the class name of the
+// parent, and stores that name in a special `_names` object on the style.
+const findAndTagDescendants = (styles, base, names={}) => {
+    Object.keys(styles).forEach(key => {
+        if (key[0] === ':' || key[0] === '@') {
+            findAndTagDescendants(styles[key], base, names);
+        } else if (key[0] === '>' && key[1] === '>') {
+            findAndTagDescendants(styles[key], base, names);
+
+            const name = `${base}__${key.slice(2)}`;
+
+            names[key.slice(2)] = {
+                _isPlainClassName: true,
+                _className: name,
+            };
+
+            styles[key]._names = { [name]: true };
```

Comment:
I'm getting a little confused here. Why is this a map if it only ever contains one key/value pair? Am I missing something here? I see above there's a traversal in `findNamesForDescendants` throughout the whole `styles` structure to find all the ancestors of a given `>>child` selector, but I don't see how it could have more than a single name in a single location here. This suspicion is supported by the fact that only a single assignment of `_names` takes place at `StyleSheet.create` time as far as I can tell.

The example above shows:

```
+ *   ">>child": {
 + *     _names: { "parent1_child_x": true, "parent2_child_y": true },
 + *     color: "red"
 + *   },
```

How did that `_names` with two keys happen given this code?